File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
import sys

input = sys.stdin.readline
n = int(input())
user = []
for i in range(n):
  age, name = input().split()
  user.append((int(age), i, name))

user.sort(key = lambda x : (x[0], x[1]))

for u in user:
  print(u[0], u[2])

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        Member[] members = new Member[n];

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int age = Integer.parseInt(st.nextToken());
            String name = st.nextToken();
            members[i] = new Member(age, name);
        }

        Arrays.sort(members, (a, b) -> a.age - b.age);

        for (int i = 0; i < n; i++) {
            System.out.println(members[i].age + " " + members[i].name);
        }
    }
}

class Member {
    int age;
    String name;

    Member(int age, String name) {
        this.age = age;
        this.name = name;
    }
}